src/uu/expr/src/syntax_tree.rs

```diff
@@ -175,96 +191,104 @@ impl StringOp {
     }
 }
 
-/// Check for errors in a supplied regular expression
-///
-/// GNU coreutils shows messages for invalid regular expressions
-/// differently from the oniguruma library used by the regex crate.
-/// This method attempts to do these checks manually in one pass
-/// through the regular expression.
-///
-/// This method is not comprehensively checking all cases in which
-/// a regular expression could be invalid; any cases not caught will
-/// result in a [ExprError::InvalidRegexExpression] when passing the
-/// regular expression through the Oniguruma bindings. This method is
-/// intended to just identify a few situations for which GNU coreutils
-/// has specific error messages.
-fn check_posix_regex_errors(pattern: &str) -> ExprResult<()> {
-    let mut escaped_parens: u64 = 0;
-    let mut escaped_braces: u64 = 0;
-    let mut escaped = false;
-
-    let mut repeating_pattern_text = String::new();
-    let mut invalid_content_error = false;
-
-    for c in pattern.chars() {
-        match (escaped, c) {
-            (true, ')') => {
-                escaped_parens = escaped_parens
-                    .checked_sub(1)
-                    .ok_or(ExprError::UnmatchedClosingParenthesis)?;
-            }
-            (true, '(') => {
-                escaped_parens += 1;
-            }
-            (true, '}') => {
-                escaped_braces = escaped_braces
-                    .checked_sub(1)
-                    .ok_or(ExprError::UnmatchedClosingBrace)?;
-                let mut repetition =
-                    repeating_pattern_text[..repeating_pattern_text.len() - 1].splitn(2, ',');
-                match (
-                    repetition
-                        .next()
-                        .expect("splitn always returns at least one string"),
-                    repetition.next(),
-                ) {
-                    ("", None) => {
-                        // Empty repeating pattern
-                        invalid_content_error = true;
+fn check_posix_regex_errors(s: &str) -> BraceContent {
```

Comment:
Did you mean to remove the doc? I wanted to be explicit about the reason why this is needed (vs seeing what errors we get from Oniguruma)